sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/CustomModelTransforms.java

```diff
@@ -67,42 +71,139 @@ static CustomFormModel toCustomFormModel(Model modelResponse) {
+            subModelList = getComposedSubmodels(modelResponse);
+            trainingDocumentInfoList = new ArrayList<>();
+            for (int i = 0; i < modelResponse.getComposedTrainResults().size(); i++) {
+                final TrainResult composedTrainResultItem = modelResponse.getComposedTrainResults().get(i);
+                final List<TrainingDocumentInfo> trainingDocumentSubModelList
+                    = composedTrainResultItem.getTrainingDocuments().stream()
+                    .map(trainingDocumentItem ->
+                        new TrainingDocumentInfo(trainingDocumentItem.getDocumentName(),
+                            TrainingStatus.fromString(trainingDocumentItem.getStatus().toString()),
+                            trainingDocumentItem.getPages(),
+                            transformTrainingErrors(trainingDocumentItem.getErrors())))
+                    .peek(trainingDocumentInfo ->
+                        PrivateFieldAccessHelper.set(trainingDocumentInfo,
+                            "modelId",
+                            composedTrainResultItem.getModelId().toString()))
+                    .collect(Collectors.toList());
+                trainingDocumentInfoList.addAll(trainingDocumentSubModelList);
+            }
         }
 
-        return new CustomFormModel(
+        CustomFormModel customFormModel = new CustomFormModel(
             modelInfo.getModelId().toString(),
             CustomFormModelStatus.fromString(modelInfo.getStatus().toString()),
             modelInfo.getCreatedDateTime(),
             modelInfo.getLastUpdatedDateTime(),
             subModelList,
             modelErrors,
             trainingDocumentInfoList);
+
+        if (modelInfo.getAttributes() != null) {
+            CustomModelProperties customModelProperties = new CustomModelProperties();
+            PrivateFieldAccessHelper.set(customModelProperties, "isComposed",
+                modelInfo.getAttributes().isComposed());
+            PrivateFieldAccessHelper.set(customFormModel, "customModelProperties",
+                customModelProperties);
+            if (modelInfo.getAttributes().isComposed()) {
+                PrivateFieldAccessHelper.set(customFormModel, "trainingDocuments",
+                    trainingDocumentInfoList);
+            }
+        }
+        if (modelInfo.getModelName() != null) {
+            PrivateFieldAccessHelper.set(customFormModel, "modelDisplayName",
+                modelInfo.getModelName());
+        }
+        return customFormModel;
+    }
+
+    /** Creates a submodel list from composed models service data **/
+    private static List<CustomFormSubmodel> getComposedSubmodels(Model modelResponse) {
+        List<CustomFormSubmodel> subModelList = new ArrayList<>();
+        for (TrainResult composedTrainResultItem : modelResponse.getComposedTrainResults()) {
+            String formType = "custom:";
+            if (modelResponse.getModelInfo().getModelName() != null) {
```

Comment:
you won't find a modelName for composed submodels since the service doesn't return this in the train result

Reply:
So for CustomFormSubmodel.formType,

If unlabeled, we keep it as `form-{clusterId}` (SDK defined)
If composed, always, `custom:{modelId}`  irrespective if modelName is present or not?
If labeled, since we only ever have one submodel, if modelName attr present then `formType = custom:{modelName}`?

Reply:
I recommend you add a test and that way you can verify that what you are doing matches what the service returns. i.e in .NET => https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormTrainingClient/FormTrainingClientLiveTests.cs#L111

Also, here is a table I did with the conversions that might help:

labeled | composed | displayname | value
-- | -- | -- | --
no | no | no | form-{clusterkey}
no | no | yes | form-{clusterkey}
yes | no | yes | custom:{displayName}
yes | no | no | custom:{modelId}
yes | yes | no | custom:{displayName} - of specific submodel
yes | yes | yes | custom:{displayName} - of specific submodel

Reply:
should value for the last two rows be changed to custom:{modelID}? If they are indicating the submodel form_type then we don't have that display name info. Ignore me if it's indicating the RecognizedForm.form_type.

Reply:
>should value for the last two rows be changed to custom:{modelID}?

I think this is correct.

Reply:
@kristapratico The submodel.formType and recognizedForm.formType, should be the same. So when a modelName is present on the labeled model (for last two rows) it should be `custom: {displayName}`, like [here](https://github.com/samvaity/azure-sdk-for-java/blob/compose_models/sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTest.java#L1307-L1314).

Reply:
Looked at the test, and agree that labeled models should have model names in form_type if present. But if enumerating the submodels on a composed model, those submodels will not have a form_type with a model_name since that info is not returned in the train result.

Reply:
updated